src/bgw/job.c

```diff
@@ -326,7 +406,7 @@ ts_bgw_job_delete_by_id(int32 job_id)
 				F_INT4EQ,
 				Int32GetDatum(job_id));
 
-	return bgw_job_delete_scan(scankey);
+	return bgw_job_delete_scan(scankey, job_id);
 }
```

Comment:
I think the comment belongs at the top of this function
```
/* This function will try to delete the job identified by `job_id`. It will send a `INT` to the job,
 * and wait for the job to terminate before returning. In the event that it cannot do this fails (for instance, if the
 * job is not in a transaction, we have no way to send the signal) this function will `ERROR`, and the user has to
 * manually determine the `pid` of the BGW and send an `INT` or a `KILL`.
 */